Source/UROSWorldControl/Private/ROSWorldControlManager.cpp

```diff
@@ -0,0 +1,73 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#include "ROSWorldControlManager.h"
+#include "Spawner.h"
+#include "Relocator.h"
+#include "Remover.h"
+
+// Sets default values
+AROSWorldControlManager::AROSWorldControlManager()
+{
+ 	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = false;
+
+}
+
+// Called when the game starts or when spawned
+void AROSWorldControlManager::BeginPlay()
+{
+	Super::BeginPlay();
+
+	UWorld* World = GetWorld();
+	if (!World) {
+		UE_LOG(LogTemp, Warning, TEXT("Couldn't find the World."));
+		return;
+	}
+	
+
+	// Setup IDMap
+	IDMap = FTagStatics::GetKeyValuesToActor(GetWorld(), "SemLog", "id");
+
+
+	// Spawn Relocator into the world
```

Comment:
you don't need to use multiple actors to listen for every type of services (the manager should subscribe to all service types)

Reply:
should I just make all the connections in the ControlManager or do you want me to but put all the functionality in there, too? I thought it was cleaner to keep everything separate, but I can see how three separate connections are not ideal.